extensions/src/main/java/org/pentaho/platform/osgi/KarafBoot.java

```diff
@@ -434,33 +463,35 @@ boolean canOpenConfigPropertiesForEdit( String directory ) {
     return true;
   }
 
-  void expandSystemPackages( String s ) {
+  void expandSystemPackages( String root ) {
+    final String customLocation = root + "/etc/custom.properties";
 
-    File customFile = new File( s );
+    File customFile = new File( customLocation );
     if ( !customFile.exists() ) {
       logger.warn( "No custom.properties file for in karaf distribution." );
       return;
     }
+
     Properties properties = new Properties();
     FileInputStream inStream = null;
     try {
       inStream = new FileInputStream( customFile );
       properties.load( inStream );
     } catch ( IOException e ) {
-      logger
-          .error( "Not able to expand system.packages.extra properties due to an error loading custom.properties", e );
+      logger.error(
+        "Not able to expand system.packages.extra properties due to an error loading custom.properties", e );
       return;
     } finally {
       IOUtils.closeQuietly( inStream );
     }
```

Comment:
This code is duplicated and can be refactored into something like a `getCustomProperties()` method that caches the properties (so the file isn't read three times: by `expandSystemPackages()`, `configureSystemProperties()` and then by `cleanCachesIfFlagSet()`). If memory is a concern, that cached version can be cleaned at the end of `startup()`.